packages/accordion/src/useAccordion.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import { useState } from 'react';
+import { composeEventHandlers, generateId, KEY_CODES } from '@zendeskgarden/container-utilities';
+
+export function useAccordion({
+  idPrefix,
+  expandedSections = [],
+  expandable = true,
+  collapsible = true
+} = {}) {
+  const [prefix] = useState(idPrefix || generateId('garden-field-container'));
+  const TRIGGER_ID = `${prefix}--trigger`;
+  const PANEL_ID = `${prefix}--panel`;
+  const [toggledSections, setToggledSections] = useState({
```

Comment:
Could this be split into 2 separate `useState`'s? If it's necessary to combine then into a single state object it might be prudent to use a `useReducer`.

Reply:
for sure! Simplified now.